### PR TITLE
Update associated pipeline during the edit flow

### PR DESCRIPTION
**Pull request by rottencandy**
**Story**:
https://issues.redhat.com/browse/ODC-5018 

**Description**: 

When there are edits made to git information like url/revision/docker file path in edit import flow, those changes should be automatically reflected in the associated pipeline as well.

**Screen shots / Gifs for design review**: 

When no pipeline is associated:
tmp0
When a pipeline is associated(checkbox is disabled):
tmp1
When the builder image is changed:
tmp2
When an unsupported builder image is selected:
tmp3
cc: @openshift/team-devconsole-ux 

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts
Comment (divyanshiGupta) [inline frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts:135]: @rottencandy when you are updating the `pipeline` with the new `template` you will need to add the `resourceVersion` of the `pipeline` in `updatedPipeline` otherwise the update request fails with the below error:
` Error "Invalid value: 0x0: must be specified for an update" for field "metadata.resourceVersion".`
```suggestion
      updatedPipeline.metadata = {
        resourceVersion: pipeline.metadata.resourceVersion,
        name: `${name}`,
        namespace,
        labels: { ...template.metadata?.labels, 'app.kubernetes.io/instance': name },
      };
```

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts
Comment (rottencandy) [reply frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts]: Updated.

**Comment by bgliwa01** [comment]
Good work! One thing - the checkboxes don't seem to be disabled...

**Comment by rottencandy** [comment]
> Good work! One thing - the checkboxes don't seem to be disabled...

@bgliwa01 They are disabled, seems like it's hard to notice in Firefox,
this is how it looks like in Chrome:
scrt-screenshot-1606759964

**Review by andrewballantyne — COMMENTED** [review]
Looking good on the surface -- will test more tomorrow

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/import/import-submit-utils.ts:482]: @rottencandy what was the reason for adding this check for `pipeline.openshift.io/runtime` label? I am asking because this wont be present in the pipeline that is associated with an app created using import from dockerfile and thus if the this app is edited, pipeline-run will not be started automatically.

Comment (rottencandy) [reply frontend/packages/dev-console/src/components/import/import-submit-utils.ts:482]: I added this check to make sure that the pipeline is still associated to the application, realized now that I should be checking for the `app.kubernetes.io/instance` label instead. Updated.

Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/import/import-submit-utils.ts]: Yes. Thanks!

File: frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts:297]: Also you don't need to add a new prop for storing the pipeline in the form state itself as if you go through `EditApplication` you will see that we are sending all the resources including pipeline to `createOrUpdateGitResources` so you can directly get the `pipeline` there.

Comment (rottencandy) [reply frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts]: Thanks, updated.

**Review by bgliwa01 — APPROVED** [review]
lgtm...based on pipelines meeting, will approve and then potentially handle changes as a bug, if needed.

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts:128]: We are starting to duplicate this label... I found 4 instances of it. Perhaps add it to `frontend/packages/pipelines-plugin/src/const.ts` and remove the duplicates.

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts:130]:
```suggestion
      updatedPipeline = _.cloneDeep(template);
```
Shouldn't need to do this

File: frontend/packages/pipelines-plugin/src/plugin.tsx
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/plugin.tsx:474]: 👍

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts
Comment (rottencandy) [reply frontend/packages/dev-console/src/components/import/import-submit-utils.ts]: Updated.

**Comment by divyanshiGupta** [comment]
Code looks good to me. Creating a cluster to test this on and will add lgtm after that.

**Comment by divyanshiGupta** [comment]
Verified it locally, works fine for me. 
/lgtm

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts
Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/import/import-submit-utils.ts]: You combined the two if statements and changed the logic. 

If we have a pipeline template, we don't want a build config. But if it's a dry run, we don't want to do Pipeline stuff because the K8s resources that support DryRun can fail. 

If it's a dry run and we have a template -- you'll request creating a BuildConfig... it's no real harm in the end, but it could theoretically fail the flow and it's just a waste of a call.

```
  if (pipeline.enabled) {
    if (!dryRun) {
      await managePipelineResources(formData, appResources);
    }
  } else {
```
and
```
  if (pipeline.enabled && !dryRun) {
    managePipelineResources(formData, appResources);
  }
  } else {
```
Are two different logic flows. 

Also, always `await` on async methods... rarely do you want to unbound a promise and let it run on it's own without context of this thread -- even if you don't want the response. If an issue happens, it can be confusing where the issue lies if you're already navigating to the Topology screen.

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts:75]:
```suggestion
  const template = _.cloneDeep(pipeline.template);
```

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

**Comment by andrewballantyne** [comment]
One major change needed - https://github.com/openshift/console/pull/7346#discussion_r534412546

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts
Comment (rottencandy) [reply frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts]: Updated.

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts
Comment (rottencandy) [reply frontend/packages/dev-console/src/components/import/import-submit-utils.ts]: Sorry, didn't think it through before combining the statements. Updated.

File: frontend/packages/pipelines-plugin/locales/en/pipelines-plugin.json
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/locales/en/pipelines-plugin.json:8]:
```suggestion
  "Pipeline will be updated to match the Builder Image.": "Pipeline will be updated to match the Builder Image.",
```
Builder Image is a resource, so a proper noun, must be title case.

**Review by andrewballantyne — APPROVED** [review]
Need to fix the Translation before this can merge so we don't need to do another PR to fix it.

**Review by andrewballantyne — APPROVED** [review]

**Review by andrewballantyne — APPROVED** [review]
/lgtm

**Pull request merged**
